src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -3566,9 +3566,7 @@ class CreateTriggerStatementSegment(BaseSegment):
 
     type = "create_trigger"
 
-    match_grammar: Matchable = Sequence("CREATE", "TRIGGER", Anything())
-
-    parse_grammar: Optional[Matchable] = Sequence(
+    match_grammar: Matchable = Sequence(
         "CREATE",
         "TRIGGER",
         Ref("TriggerReferenceSegment"),
```

Comment:
This is the clean up @barrywhart 

The `match_grammar` does nothing the `parse_grammar` doesn't (and didn't work in TSQL). So collapsed into one. Same in postgres.

Reply:
Cool. In terms of the type hints and the new check, it's cleaner if the ANSI dialect only provides one of the two.